sotopia/generation_utils/generate.py

```diff
@@ -391,7 +391,7 @@ def format_bad_output_for_script(
     ill_formed_output: str,
     format_instructions: str,
     agents: list[str],
-    model_name: str = "gpt-3.5-turbo",
+    model_name: str = "gpt-4o-mini",
```

Comment:
I am worried that would be deprecated soon. So to keep consistent with the previous convention to use the "turbo" model without a specific date, let's use "gpt-4o-mini" to clear running problems for now.

In general, I think for `agenerate()` method, we should allow the user to specify their own backup model choice -- but that would mean a bigger change. I just want to keep this PR as small as possible to clear possible urgent problems.

Reply:
Yeah, I would say this is incrementally better, since 3.5-turbo was not a fixed version either. @yangalan123 Is preparing another feature which exposes the model of `format_bad_output` to `agenerate`

Reply:
just to note that `3.5-turbo` is a fixed version. We should probably get rid of that at some point

https://github.com/sotopia-lab/sotopia/blob/1d35c78d524490332d0aae8df718c4cd9ae4aedd/sotopia/generation_utils/generate.py#L280

Reply:
In general, I would just push for the fixed model version for reproducibility reasons. However, probably nothing is reproducible anyway for openai models. So I am okay with just doing 4o-mini

Reply:
@XuhuiZhou I see your concern on reproducibility -- however, as I heard GPT4-Next is around the corner, we might soon need to return to a similar issue as that is guaranteed to be a large release if we anchor `GPT-4o-mini` to some specific versions (and for other non-finetuned models). In general, to make Sotopia helpful and useful, we should just keep those outdated configs somewhere (for recording purposes of your papers) and allow the user to specify their own configurations. Users can make their experiments reproducible by providing detailed configs for main models, backup models, etc. Moving forward, I do not think Sotopia developers can or should take this responsibility for users (but great thanks for maintaining that level of reproducibility so far!). So after my next PR on allowing the user to specify their own preferred version of models for handling bad-formatted outputs, we should delete that `_return_fixed_model_version` in my mind to make this project more scalable and flexible. Another way of thinking about this problem is that, we should not make decisions on behalf of users on which version they should stick to.